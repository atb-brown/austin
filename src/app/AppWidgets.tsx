import Counter from "../component/widget/count/Counter";
import GuestBook from "../component/widget/guestBook/GuestBook";
import React from "react";

/**
 * This component holds widgets with which I'm experimenting with React.
 *
 * @return {ReactElement}
 */
export default function AppWidgets(): JSX.Element {
  return (
    <div
      style={{
        alignItems: "center",
        display: "flex",
        flex: 1,
        flexDirection: "column",
        gap: 10,
        justifyContent: "center",
        width: "100%",
      }}
    >
      <div
        style={{
          alignItems: "center",
          display: "flex",
          flex: 1,
          flexDirection: "column",
          justifyContent: "center",
          width: "100%",
        }}
      >
        <Counter />
      </div>
      <div
        style={{
          alignItems: "center",
          display: "flex",
          flex: 8,
          flexDirection: "column",
          justifyContent: "center",
          width: "95%",
        }}
      >
        <GuestBook />
      </div>
    </div>
  );
}
