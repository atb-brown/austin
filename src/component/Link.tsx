import React from "react";

/**
 * A basic, clickable link for a URL.
 *
 * @param {LinkProps} props Properties.
 * @return {ReactElement} The link component.
 */
export default function Link(props: LinkProps) {
  return (
    <a className="App-link" href={props.url}>
      {props.description}
    </a>
  );
}

type LinkProps = {
  description: string;
  url: string;
};
