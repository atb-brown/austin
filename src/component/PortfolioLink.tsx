import Link from "./Link";
import React from "react";

export default function PortfolioLink(props: LinkProps) {
  if (props.issuerDescription && props.issuerUrl) {
    return (
      <LinkWithIssuer
        description={props.description}
        url={props.url}
        urlId={props.urlId}
        issuerDescription={props.issuerDescription}
        issuerUrl={props.issuerUrl}
      />
    );
  }
  return (
    <LinkWithoutIssuer
      description={props.description}
      url={props.url}
      urlId={props.urlId}
    />
  );
}

function LinkWithoutIssuer(props: LinkProps) {
  // TODO: it's a bad idea to have <div/> elements that are children of <p/> elements
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <br />
    </div>
  );
}

function LinkWithIssuer(props: LinkProps) {
  return (
    <div>
      <div>{props.description}</div>
      <div>
        <Link description={props.urlId} url={props.url} />
      </div>
      <div className="App-soft-text">
        Issued by:{" "}
        <Link
          description={props.issuerDescription ?? ""}
          url={props.issuerUrl ?? ""}
        />
      </div>
      <br />
    </div>
  );
}

type LinkProps = {
  description: string;
  url: string;
  urlId: string;
  issuerDescription?: string;
  issuerUrl?: string;
};
