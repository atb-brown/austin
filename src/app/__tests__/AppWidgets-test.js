import { wait as guestBook } from "../../test/component/GuestBookTestUtils";
import { registerMockProvider as mockGuestBookProvider } from "../../test/mock/hook/mockUseGuestBookInfo";
import AppWidgets from "../AppWidgets";
import { render } from "@testing-library/react";

it("renders to match snapshot", async () => {
  mockGuestBookProvider();
  const renderResult = render(<AppWidgets />);

  await guestBook(renderResult);

  expect(renderResult.asFragment()).toMatchSnapshot();
});
