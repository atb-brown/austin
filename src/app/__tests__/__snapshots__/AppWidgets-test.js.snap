// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    style="align-items: center; display: flex; flex: 1; flex-direction: column; gap: 10px; justify-content: center; width: 100%;"
  >
    <div
      style="align-items: center; display: flex; flex: 1; flex-direction: column; justify-content: center; width: 100%;"
    >
      <div
        class="disable-text-selection"
        data-testid="counter"
        style="align-items: center; background-color: steelblue; border-radius: 10px; display: flex; font-size: 20px; height: 60px; justify-content: center; width: 25%;"
        title="This component counts the number of times it has been clicked. It's an exercise in handling clicks and component state!"
      >
        0
      </div>
    </div>
    <div
      style="align-items: center; display: flex; flex: 8; flex-direction: column; justify-content: center; width: 95%;"
    >
      <div
        data-testid="guest-book"
        style="display: flex; flex-direction: column; gap: 3px; min-height: 50%; width: 99%;"
        title="This component displays everyone who has signed my guest book. It's an exercise in HTTP requests and dependency injection!"
      >
        <div
          style="border-bottom-left-radius: 15px; border-top-left-radius: 15px; border-top-right-radius: 15px; display: flex; height: 85px; overflow: hidden; width: 100%;"
        >
          <div
            class="Guests-Visual"
            style="flex: 2; flex-direction: column; justify-content: space-evenly;"
          >
            <div
              class="Guests-Text-container"
              style="font-size: 75%;"
            >
              austin
            </div>
            <div
              class="Guests-Text-container"
              style="font-size: 50%; justify-content: center;"
            >
              www.example.com
            </div>
          </div>
          <div
            class="Guests-Visual Guests-Text-Container"
            style="flex: 7; margin-left: 3px; margin-right: 3px;"
          >
            <div
              class="Guests-Text-Container"
              style="font-size: 75%; height: 100%; width: 98%;"
            >
              info loaded
            </div>
          </div>
          <div
            class="Guests-Visual Guests-Text-Container"
            style="flex: 1; font-size: 60%;"
          >
            2023
          </div>
        </div>
        <a
          href="https://github.com/atb-brown/guest-book/edit/main/guests.json"
          style="align-items: center; align-self: flex-end; background-color: rebeccapurple; border-bottom-left-radius: 15px; border-bottom-right-radius: 15px; border-top-left-radius: 15px; color: white; display: flex; font-size: 80%; height: 85px; justify-content: center; text-decoration: none; width: 15%;"
        >
          Sign my guest book!
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
